nym-vpn-core/crates/nym-vpn-lib-types/src/tunnel_state.rs

```diff
@@ -198,6 +205,117 @@ pub enum ErrorStateReason {
     Internal(String),
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ClientErrorReason {
+    Firewall,
+    Routing,
+    SameEntryAndExitGateway,
+    InvalidEntryGatewayCountry,
+    InvalidExitGatewayCountry,
+    MaxDevicesReached,
+    BandwidthExceeded,
+    SubscriptionExpired,
+    Dns(String),
+    Api(String),
+    Internal(String),
```

Comment:
Since in the underlying tunnel error system, the 3 equivalent variants details are optional, I think it would be better to reflect that. Instead of going strict (and returning a runtime error if no details), we should allow empty details:

```
    Dns(Option<String>),
    Api(Option<String>),
    Internal(Option<String>),

// [...]

impl From<ProtoErrorStateReason> for ErrorStateReason {
// [...]
            ErrorStateReason::Dns => Ok(ClientErrorReason::Dns(value.detail)),
            ErrorStateReason::Api => Ok(ClientErrorReason::Api(value.detail)),
            ErrorStateReason::Internal => Ok(ClientErrorReason::Internal(value.detail)),
```